server/src/main/java/org/apache/druid/guice/http/JettyHttpClientModule.java

```diff
@@ -41,6 +51,7 @@
 public class JettyHttpClientModule implements Module
 {
   private static final long CLIENT_CONNECT_TIMEOUT_MILLIS = TimeUnit.MILLISECONDS.toMillis(500);
+  private static QueuedThreadPool httpClientThreadPool = null;
```

Comment:
This is not safe to store globally. Each client might have its own pool

Reply:
there's only one server per service, so while not great to store it as a global variables, it's probably safe on the server side. For clients I don't think we can assume only one client per node will exist.